### PR TITLE
Config repos user docs

### PR DESCRIPTION
**Pull request by tomzo**
* XML reference of the `<config-repos>` section with some examples
 * user docs about config-repos extension point, including:
  * overview of feature
  * scope of what can be defined in remote pipelines
  * what merging behavior is expected for groups and environments
  * errors and limitations

### CODE REVIEW

File: configuration/configuration_reference.md [hunk 1]
```diff
@@ -562,6 +599,85 @@ Two users would be administrators, they are **Jez** and **lqiao**.
...
+## &lt;config-repos&gt; {#config-repos}
+
+The `<config-repos>` element is a container of many `<config-repo>`.
+
+### Example
+
+```xml
+<cruise>
+  ...
+  <config-repos>
```
Comment (zabil) [inline configuration/configuration_reference.md]: Can we make this example use the default config repo behaviour without plugins i.e. xml configuration?
It's not clear (in this section) that these plugins have to be installed separately.

Comment (arvindsv) [inline configuration/configuration_reference.md]: The XML config repo thing is not going to be public. It's not a real plugin. It has been removed from the original doc as well (the google drive one).

File: extension_points/configrepo_extension.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# Configuration repository Extension
+
+Go supports writing configuration plugins starting `16.7`.
```
Comment (arvindsv) [inline extension_points/configrepo_extension.md]: GoCD?

File: extension_points/configrepo_extension.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+It is a feature which allows you to move pipeline configurations out of GoCD and its cruise-config.xml file into one or more source-control repositories (e.g git), so that you can modify them externally. Such modifications will be seen by a periodic poller in the GoCD server and it will [merge](#merging-configurations) those pipeline configurations into the pipelines it finds in the main configuration XML file. You can see a quick preview video of this feature [here](https://www.youtube.com/watch?v=1AfBxCWRqD8).
```
Comment (zabil) [inline extension_points/configrepo_extension.md]: (e.g git) to (e.g. git)
I think it's ok to not have the link to the video the description is clear. The video is long and is titled as "Preview"

File: extension_points/configrepo_extension.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+ - in repository A we can define that `pipeline1` is member of environment `development`
+ - in repository B we can define that `pipeline2` and `pipeline3` is member of environment `development`
+ - in repository C we can define that `pipeline3` is member of environment `development`
+
+Then in Go server, the final pipelines in environment `development` are `pipeline1`, `pipeline2`, `pipeline3`.
+Notice that `pipeline3` membership in environment `development` was declared twice.
+
+Same approach is used for agents.
+
+Environment variables have additional check. If you assign same environment variable
```
Comment (zabil) [inline extension_points/configrepo_extension.md]: Environment variables have additional check(s) ? 
or
Environment variables have an additional check?

**Comment by tomzo** [comment]
@zabil thanks for review. I updated the minor parts that you pointed.

**Pull request merged**